### PR TITLE
Use calculate

### PR DESCRIPTION
**Pull request by elsaperelli**
# Summary
This PR replaces uses of `calculateMeasureReports` with `calculate` from `fqm-execution`. Information is now pulled off of the `detailedResults` output rather than the `MeasureReport`. This will make things easier in the future when we want individual episode results from the detailedResults.

## New Behavior
- None. The UI should look and act the same.

## Code Changes
- `utils/types.ts` - Added new `DetailedResult` type to make things easier. In the future, we can export these types from `fqm-execution` differently, but this works for now.
- `util/MeasurePopulations.ts` - Added `populationCriteriaExpression` since that is actually where we get the table headers from now since we are pulling population results from detailedResults rather than the Measure Report. Should I remove `populationCode` and `populationDisplay`?
- `util/MeasureCalculation` - Should I rename this file? Changes `calculateMeasureReport` to `calculateDetailedResult`.
- `measureReportLookup.ts` -> `detailedResultLookup.ts` 
- `pages/generate-test-cases.tsx` / `components/resource-creation/ResourcePanel.tsx` / `component/resource-creation/ResourceDisplay.tsx` / `components/patient-creation/PatientCreationPanel.tsx` / `components/calculation/PopulationCalculation.tsx` - calculateMeasureReports to calculateDetailedResults switcheroos
- `MeasureHighlightingPanel.tsx` - logic highlighting comes off of `detailedResults.html` property now
- `PopulationComparisonTable.tsx` - Gets the key from the `criteriaExpression` now 
- `PopulationResultsTable.tsx` - `ecqm-visualizers` takes in a MeasureReport, so I decided to just bring in the code from that repo in here and change it to take in detailedResults. Let me know if I missed anything. Sorry `ecqm-visualizers` :(
- `package.json` / `package-lock.json` - Removed `ecqm-visualizers` and added the dependencies we need that came from it
- `PopulationCalculation.test.tsx` / `PopulationComparisonTable.test.tsx` - fixed affected unit tests

# Testing Guidance
- `npm run check`
- `npm run dev` and navigate to http://localhost:3000/
- Make sure the app runs and looks the same!

NOTE: There are some errors in the unit tests that are not being picked up. Nat fixes them in #63, so just ignore them for now.

### CODE REVIEW

File: components/calculation/PopulationResultsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { useMemo } from 'react';
+import { DetailedResult } from '../../util/types';
+
+export type PopulationResultViewerProps = {
+  results: DetailedResultInfoArray;
+};
+
+export type DetailedResultInfoArray = Array<DetailedDetailedResult>;
+
+export declare type DetailedDetailedResult = {
```
[the remark this answers is not in the document]
Comment (natjoe4) [reply components/calculation/PopulationResultsTable.tsx:11]: Also, just curious... What does `declare` do here? Haven't seen that before

File: components/calculation/PopulationResultsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    const group = dr.detailedResults?.[0];
+    const populationResults = { label };
+    group?.populationResults?.reduce((acc: PopulationResult, e) => {
+      const key = e.criteriaExpression || e.populationType;
+      if (key) {
+        acc[key as string] = e.result === true ? 1 : 0;
+      }
+      return acc;
+    }, populationResults);
+    return PopulationResultsRow(populationResults);
```
Comment (natjoe4) [inline components/calculation/PopulationResultsTable.tsx:50]: The term `populationResults` is super overloaded here, consider changing the variable to `labeledPopulationResults` or something?

**Review by natjoe4 — CHANGES_REQUESTED** [review]
Functionality seems unchanged to me, so great job catching all the effects of this change. Just a few smaller comments :)

File: components/calculation/PopulationCalculation.tsx [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
 import { useState } from 'react';
-import { fhirJson } from '@fhir-typescript/r4-core';
 import { measurementPeriodState } from '../../state/atoms/measurementPeriod';
 import { showNotification } from '@mantine/notifications';
 import { IconAlertCircle } from '@tabler/icons';
 import Link from 'next/link';
 import { getPatientInfoString } from '../../util/fhir/patient';
 import { createPatientBundle } from '../../util/fhir/resourceCreation';
+import { DetailedDetailedResult } from './PopulationResultsTable';
+import PopulationResultTable from './PopulationResultsTable';
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply components/calculation/PopulationCalculation.tsx:14]: Smooshed ✅

File: components/calculation/PopulationResultsTable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (elsaperelli) [reply components/calculation/PopulationResultsTable.tsx:11]: Hahahah you mean to say `DetailedDetailedResult` doesn't look or sound good?? And actually I am also not sure- I think I may have pulled that from `ecqm-visualizers`. I think I can remove it.

File: components/calculation/PopulationCalculation.tsx [hunk 4]
```diff
@@ -68,29 +69,28 @@ export default function PopulationCalculation() {
...
+      .then(detailedResults => {
+        if (detailedResults) {
           const patientLabels = createPatientLabels();
-          const labeledMeasureReports: DetailedMeasureReport[] = [];
-          measureReports.forEach(mr => {
-            const patientId = mr.subject?.reference?.split('/')[1];
-            labeledMeasureReports.push({
+          const labeledDetailedResults: DetailedDetailedResult[] = [];
+          detailedResults.forEach(dr => {
+            const patientId = dr.patientId;
```
[the remark this answers is not in the document]
Comment (elsaperelli) [reply components/calculation/PopulationCalculation.tsx:87]: It probably doesn't need to be an interface then, right?

**Review by natjoe4 — APPROVED** [review]
Looks great! Nice work :)

**Review by hossenlopp — APPROVED** [review]
Excellent work!

**Pull request merged**